core/src/main/java/com/linecorp/armeria/server/VirtualHostBuilder.java

```diff
@@ -1524,6 +1513,29 @@ private SelfSignedCertificate selfSignedCertificate() throws CertificateExceptio
         return selfSignedCertificate;
     }
 
+    boolean equalsDefaultHostname(String defaultHostname) {
+        return normalizeDefaultHostname(defaultHostname).equals(this.defaultHostname);
+    }
+
+    boolean equalsHostnamePattern(String hostnamePattern) {
```

Comment:
What do you think of parsing `hostnamePattern` before calling this method? Otherwise, the same `hostnamePattern` will be parsed as many virtual hosts.

```suggestion
    boolean equalsHostnamePattern(String validhostnamePattern, int port) {
```

Reply:
@ikhoon I applied your comment as I understand. PTAL 🙏